thinc/loss.py

```diff
@@ -47,12 +49,15 @@ def __init__(
         missing_value: Optional[Union[str, int]] = None,
         neg_prefix: Optional[str] = None,
         label_smoothing: float = 0.0,
+        class_weights: Optional[List[float]] = None,
     ):
         self.normalize = normalize
         self.names = names
         self.missing_value = missing_value
         self.neg_prefix = neg_prefix
         self.label_smoothing = label_smoothing
+        self.class_weights = class_weights
```

Comment:
Maybe the list -> array conversion should be done here already? Avoids allocating/freeing the same array repeatedly.

Reply:
I was thinking the same but we don't know whether we will get `cupy` or `numpy` later on.

Reply:
What do you think about the current solution @danieldk ?

Reply:
Hmm, then we'd still reallocate it over and over again if the user passes a list. What about something more drastic, like just requiring `class_weights` to be `Floats1d`?

Reply:
Yeah ok we can do that too, but then still since the `CategoricalCrossEntropy` itself is not bound to be `cupy` or `numpy` and it could get a different type of array at every call, we would still need to reallocate right?

Reply:
Right, but I don't think that happens much in practice. Probably when someone passes in `class_weights` as a CuPy array they'll also use CuPy arrays down the line. I think at this point it probably doesn't make much sense to optimize for mixed use.